specs/create-rbd-spec.adoc

```diff
@@ -0,0 +1,236 @@
+= Create RBD workflow by using tendrl-frontend UI
+
+The intent of create RBD workflow is to provide a mechanism to create one or more Ceph RBDs into tendrl for management using tendrl-frontend UI.
+
+* Assumption: 
+
+    - Ceph cluster exists.
+
+    - Ceph pool in the Ceph cluster may or may exist.
+
+== Problem description
+
+This specification describes a mechanism to create one or more Ceph RBDs into tendrl for management.
+
+== Use Cases
+
+This addresses the use case of creating RBD into tendrl for management.
+
+The workflow is as follows:
+
+* "Create RBD" is triggered by "Create RBD" button on RBD landing page.
+   (https://redhat.invisionapp.com/share/BR8JDCGSQ#/screens/198417119)
+
+* User clicks on "Create" button.
+
+* It will redirect the user to a wizard for creating RBD/RBDs.
+
+* The flow consist of three steps.
+
+* In the first step, user needs to fill below items to the form -
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735641)
+
+    - RBD Name
+
+    - RBDs to create(UI will autogenerate the name of RBDs based on the number)
+
+    - cluster(User will select from the drop down which will list all the clusters present in Tendrl)
+
+    - Target size
+
+* If the target size * RBDs to create > available storage in this cluster, a warning message will be displayed, but 
+  the user will be allowed to continue.
+
+* For block devices, a backing pool must be created. Only one pool can be created per set of block devices.
+
+* In the second step, user will decide whether to use an existing pool or create a new pool as the backing pool
+  for this set of RBD.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735638)
+
+* On choosing existing pool, a drop down will be displayed which will list down all the available pools in the 
```

Comment:
It is not possible to create an RBD image on an erasure coded pool because it requires partial writes. It is however possible to create an RBD image on an erasure coded pools when a replicated pool tier set a cache tier.  See http://docs.ceph.com/docs/master/rados/operations/erasure-code/.  Agree that only replicated pools get listed.